src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -307,7 +307,14 @@ def __init__(self, secret_store):
         self._entries = []
 
     def load_entry(self, sp_id, tenant):
+        from azure.cli.core._profile import env_var_auth_configured, load_env_var_credential
         self._load_persistence()
+
+        # If no login data, look for service principal credential in environment variables
+        if not self._entries and env_var_auth_configured():
+            logger.warning("Using service principal credential configured in environment variables.")
+            self._entries = [load_env_var_credential()]
```

Comment:
`load_entry` is designed solely for looking up service principal credentials stored on the hard disk. It may not be an ideal place for loading environment variables.